src/Microsoft.ML.OnnxTransform/OnnxTransform.cs

```diff
@@ -129,33 +154,52 @@ private OnnxTransform(IHostEnvironment env, Arguments args, byte[] modelBytes =
-            OutputType = new VectorType(type, dims);
+            //if (modelInfo.InputsInfo.Length != 1)
+            //    throw env.Except($"OnnxTransform supports Onnx models with one input. The provided model has ${modelInfo.InputsInfo.Length} input(s).");
+            //if (modelInfo.OutputsInfo.Length != 1)
+            //    throw env.Except($"OnnxTransform supports Onnx models with one output. The provided model has ${modelInfo.OutputsInfo.Length} output(s).");
+
+            Inputs = args.InputColumns;
+            Outputs = args.OutputColumns;
+            //var type = OnnxUtils.OnnxToMlNetType(outputNodeInfo.Type);
+            //var shape = outputNodeInfo.Shape;
+            //var dims = shape.Count > 0 ? shape.Skip(shape[0] < 0 ? 1 : 0).Select( x => (int) x ).ToArray() : new[] { 0 };
+
+            OutputTypes = new ColumnType[args.OutputColumns.Length];
+
+            var numModelOutputs = Model.ModelInfo.OutputsInfo.Length;
+            for (int i=0; i < args.OutputColumns.Length; i++)
+            {
+                var idx = -1;
+                for (var j = 0; j < Model.ModelInfo.OutputsInfo.Length; j++)
+                    if (Model.ModelInfo.OutputsInfo[j].Name == args.OutputColumns[i])
+                    {
+                        idx = j;
+                        break;
+                    }
+                if (idx < 0)
+                    throw _host.Except($"Column {args.OutputColumns[i]} doesn't match output node names of model");
+                var outputNodeInfo = Model.ModelInfo.OutputsInfo[idx];
+                var shape = outputNodeInfo.Shape;
+                var dims = shape.Count > 0 ? shape.Skip(shape[0] < 0 ? 1 : 0).Select(x => (int)x).ToArray() : new[] { 0 };
+                OutputTypes[i] = new VectorType(OnnxUtils.OnnxToMlNetType(outputNodeInfo.Type), dims);
+            }
             _args = args;
         }
 
-        public OnnxTransform(IHostEnvironment env, string modelFile, string inputColumn, string outputColumn)
-            : this(env, new Arguments() { ModelFile = modelFile, InputColumn = inputColumn, OutputColumn = outputColumn })
+        public OnnxTransform(IHostEnvironment env, string modelFile, string[] inputColumns, string[] outputColumns)
+            : this(env, new Arguments() { ModelFile = modelFile, InputColumns = inputColumns, OutputColumns = outputColumns })
         {
         }
 
         public Schema GetOutputSchema(Schema inputSchema)
         {
             _host.CheckValue(inputSchema, nameof(inputSchema));
-            if (!inputSchema.TryGetColumnIndex(Input, out int srcCol))
-                throw _host.ExceptSchemaMismatch(nameof(inputSchema), "input", Input);
-
+            foreach (var input in Inputs)
+            {
+                if (!inputSchema.TryGetColumnIndex(input, out int srcCol))
+                    throw _host.ExceptSchemaMismatch(nameof(inputSchema), "input", input);
```

Comment:
>ExceptSchemaMismatch [](start = 32, length = 20)

At what place we check type of input column? during transform?

Reply:
We check it in the other function GetOutputSchema, at line ~500. We check for shape mismatch, and type mismatch as well.

---
In reply to: [233269217](https://github.com/dotnet/machinelearning/pull/1586#discussion_r233269217) [](ancestors = 233269217)